pkg/httphelper/response_writer.go

```diff
@@ -0,0 +1,63 @@
+package httphelper
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/net/context"
+)
+
+func NewResponseWriter(w http.ResponseWriter) *ResponseWriter {
+	return &ResponseWriter{w: w, status: 0}
+}
+
+type ResponseWriter struct {
+	Context context.Context
```

Comment:
unexport this and use a method to export it, in case we want to use this as an interface value.